File: app/src/main/java/com/flybits/conciergesample/activity/java/PushHandleActivity.java
package com.flybits.conciergesample.activity.java;

import static com.flybits.android.push.services.PushServiceKt.EXTRA_PUSH_NOTIFICATION;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.flybits.concierge.enums.ConciergePush;
import com.flybits.conciergesample.R;

import com.flybits.concierge.Concierge;

public class PushHandleActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_push_handle);

        final FragmentManager supportFragmentManager = getSupportFragmentManager();
        final FragmentTransaction transaction = supportFragmentManager.beginTransaction();
        final Intent intent = getIntent();
        if (intent != null) {
            if (intent.hasExtra(EXTRA_PUSH_NOTIFICATION)) {
                final ConciergePush push = Concierge.INSTANCE.handlePush(this, intent);
                if (push != null) {
                    final Fragment fragment = Concierge.INSTANCE.deepLink(push, this, null);
                    if (fragment != null) {
                        transaction.replace(R.id.fragment_container, fragment);
                    }
                }
            }
            transaction.commit();
        }
    }
}

File: app/src/main/java/com/flybits/conciergesample/activity/PushHandleActivity.kt
package com.flybits.conciergesample.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.flybits.android.push.services.EXTRA_PUSH_NOTIFICATION
import com.flybits.concierge.Concierge
import com.flybits.conciergesample.R

class PushHandleActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_push_handle)

        val transaction = supportFragmentManager.beginTransaction()
        intent?.let {
            if (it.hasExtra(EXTRA_PUSH_NOTIFICATION)) {
                val push = Concierge.handlePush(this, it)
                if (push != null) {
                    Concierge.deepLink(push, this)?.let { fragment ->
                        transaction.replace(R.id.fragment_container, fragment)
                    }
                }
            }
            transaction.commit()
        }
    }
}